.github/workflows/downstream-project-spl.yml

```diff
@@ -0,0 +1,80 @@
+name: Downstream Project - SPL
+
+on:
+  push:
+    branches:
+      - master
+      - v[0-9]+.[0-9]+
+  pull_request:
+    branches:
+      - master
+      - v[0-9]+.[0-9]+
+    paths:
+      - "**.rs"
+      - "Cargo.toml"
+      - "Cargo.lock"
+      - "cargo-build-bpf"
+      - "cargo-test-bpf"
+      - "cargo-build-sbf"
+      - "cargo-test-sbf"
+      - "ci/downstream-projects/run-spl.sh"
+      - ".github/workflows/downstream-project-spl.yml"
+  workflow_call:
+    inputs:
+      branch:
+        required: false
+        type: string
+        default: "master"
+
+env:
+  SHELL: /bin/bash
+  SCCACHE_GHA_ENABLED: "true"
+  RUSTC_WRAPPER: "sccache"
+
+jobs:
+  main:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        program:
+          - instruction-padding/program
+          - token/program
+          - token/program-2022
+          - token/program-2022-test
+          - associated-token-account/program
+          - token-upgrade/program
+          - feature-proposal/program
+          - governance/addin-mock/program
+          - governance/program
+          - memo/program
+          - name-service/program
+          - stake-pool/program
+          - stake-pool/single-pool
```

Comment:
Rather than specifying each program one-by-one and then building the whole workspace, it should be faster to specify chains of dependent programs, and remove the `$CARGO_BUILD_SBF` step.

Here are the dependent chains:

* token/program
* instruction-padding/program, token/program-2022, token/program-2022-test <-- we can eventually move out instruction-padding into its own, but not yet
* associated-token-account/program, associated-token-account/program-test
* token-upgrade/program
* feature-proposal/program
* governance/addin-mock/program, governance/program
* memo/program
* name-service/program
* stake-pool/program
* stake-pool/single-pool

Reply:
I had some changes here: https://github.com/solana-labs/solana/pull/32677/files#diff-8a9809831c9c46b49c232e9735043e71b17915276e8cbf50d3b14c12dab3f19aR62-R103